Game/LambdaHack/Server/Commandline.hs

```diff
@@ -9,162 +12,277 @@ import Game.LambdaHack.Common.Prelude
+debugModeSerP = do
+  knowMap              <- knowMapP
+  knowEvents           <- knowEventsP
+  knowItems            <- knowItemsP
+  sniffIn              <- sniffInP
+  sniffOut             <- sniffOutP
+  allClear             <- allClearP
+  boostRandomItem      <- boostRandItemP
+  gameMode             <- gameModeP
+  automateAll          <- automateAllP
+  keepAutomated        <- keepAutomatedP
+  ~(newGame,challenge) <- serToChallenge <$> newGameP
+  stopAfterSeconds     <- stopAfterSecsP
+  stopAfterFrames      <- stopAfterFramesP
+  benchmark            <- benchmarkP
+  dungeonRng           <- setDungeonRngP
+  mainRng              <- setMainRngP
+  dumpInitRngs         <- dumpInitRngsP
+  dbgMsgSer            <- dbgMsgSerP
+  gtkFontFamily        <- gtkFontFamilyP
+  sdlFontFile          <- sdlFontFileP
+  sdlTtfSizeAdd        <- sdlTtfSizeAddP
+  sdlFonSizeAdd        <- sdlFonSizeAddP
+  fontSize             <- fontSizeP
+  noColorIsBold        <- noColorIsBoldP
+  maxFps               <- maxFpsP
+  disableAutoYes       <- disableAutoYesP
+  noAnim               <- noAnimP
+  savePrefix           <- savePrefixP
+  frontendTeletype     <- frontendTeletypeP
+  frontendNull         <- frontendNullP
+  frontendLazy         <- frontendLazyP
+  dbgMsgCli            <- dbgMsgCliP
+
+  pure DebugModeSer
+    { sknowMap         = knowMap
+    , sknowEvents      = knowEvents
+    , sknowItems       = knowItems
+    , sniffIn          = sniffIn
+    , sniffOut         = sniffOut
+    , sallClear        = allClear
+    , sboostRandomItem = boostRandomItem
+    , sgameMode        = gameMode
+    , sautomateAll     = automateAll
+    , skeepAutomated   = keepAutomated
+    , sdungeonRng      = dungeonRng
+    , smainRng         = mainRng
+    , snewGameSer      = newGame
+    , scurChalSer      = challenge
+    , sdumpInitRngs    = dumpInitRngs
+    , ssavePrefixSer   = savePrefix
+    , sdbgMsgSer       = dbgMsgSer
+    , sdebugCli        = defDebugModeCli
+        { sgtkFontFamily    = gtkFontFamily
+        , sdlFontFile       = sdlFontFile
+        , sdlTtfSizeAdd     = sdlTtfSizeAdd
+        , sdlFonSizeAdd     = sdlFonSizeAdd
+        , sfontSize         = fontSize
+        , scolorIsBold      = noColorIsBold
+        , smaxFps           = maxFps
+        , sdisableAutoYes   = disableAutoYes
+        , snoAnim           = noAnim
+        , snewGameCli       = newGame
+        , sbenchmark        = benchmark
+        , ssavePrefixCli    = savePrefix
+        , sfrontendTeletype = frontendTeletype
+        , sfrontendNull     = frontendNull
+        , sfrontendLazy     = frontendLazy
+        , sdbgMsgCli        = dbgMsgCli
+        , sstopAfterSeconds = stopAfterSeconds
+        , sstopAfterFrames  = stopAfterFrames
+        }
+    }
+ where
+   serToChallenge :: Maybe Int -> (Bool, Challenge)
+   serToChallenge Nothing  = (False, defaultChallenge)
+   serToChallenge (Just n) = (True, Challenge n False False)
```

Comment:
You unravelled it correctly, even though the original code suggested challenge can be modified by other options (which was not in fact the case). BTW, I'd use `defaultChallenge` also in this line, for consistency and clarity (`cdiff` says "this Int represents difficulty"), e.g., `(Just cdiff) = (True, defaultChallenge {cdiff})`